lib/utils/logDeprecation.js

```diff
@@ -55,22 +55,29 @@ const resolveMode = (serviceConfig) => {
   }
 };
 
-function writeDeprecation(code, message) {
+function writeDeprecation(code, message, { onlyLegacy } = {}) {
   const prefix = 'Serverless: ';
   const messageLines = message.split('\n');
+  const modernLogsMessageLines = [...messageLines];
   const followingLinesPrefix = ' '.repeat(prefix.length);
   for (let i = 1; i < messageLines.length; ++i) {
     messageLines[i] = followingLinesPrefix + messageLines[i];
   }
   message = messageLines.join('\n');
   legacy.write(`Serverless: ${chalk.keyword('orange')(`Deprecation warning: ${message}`)}\n`);
   if (!code.startsWith('EXT_')) {
+    modernLogsMessageLines.push(
+      `More Info: https://www.serverless.com/framework/docs/deprecations/#${code}`
+    );
     legacy.write(
       `            ${chalk.keyword('orange')(
         `More Info: https://www.serverless.com/framework/docs/deprecations/#${code}`
       )}\n`
     );
   }
+  if (!onlyLegacy) {
+    log.warning(modernLogsMessageLines.join('\n'));
```

Comment:
It'll be nicer to go with what you proposed earlier, but just add this condition here as: `if (isModern) log.warning`

It'll also make update minimal in v3. 

(Passing it through option in internal function, where answer is visible upfront looks a bit overkill)

Reply:
Indeed, you're right. Let's keep it the way you've proposed, sorry for confusion